torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -297,11 +297,13 @@ void FoldPrePackingOps(script::Module& m) {
 }
 
 void optimizeForMobile(script::Module& m) {
-  m.eval();
-  m = FoldConvBatchNorm2d(m);
-  insertPrePackedOps(m);
-  m = freeze_module(m);
-  FoldPrePackingOps(m);
+  auto cloned_module = module.clone();
+  cloned_module.eval();
+  cloned_module = FoldConvBatchNorm2d(cloned_module);
+  insertPrePackedOps(cloned_module);
+  cloned_module = freeze_module(cloned_module);
+  FoldPrePackingOps(cloned_module);
+  return cloned_module;
```

Comment:
This function's return type is void, right?  You are probably getting a compiler warning which PyTorch's compilation settings do not treat as error.  Also if we are no longer modifying the parameter please pass it by const reference.

Reply:
It's still safer to pass the argument by const ref.  Please consider that if there are no other complications.  Otherwise we wouldn't have a way of knowing if we accidentally modify m.